conan/cli/formatters/report/diff.py

```diff
@@ -0,0 +1,99 @@
+import json
+import os
+import base64
+
+from jinja2 import Template
+
+from conan.api.output import cli_out_write
+from conan.cli.formatters.report.diff_html import diff_html
+
+def _generate_json(result):
+    diff_text = result["diff"]
+    src_prefix = result["src_prefix"]
+    dst_prefix = result["dst_prefix"]
+    ret = {}
+    current_filename = None
+    for line in diff_text.splitlines():
+        if line.startswith("diff --git "):
+            src_filename, dst_filename = _get_filenames(line, src_prefix, dst_prefix)
+            current_filename = src_filename
+            ret[current_filename] = [line]
+        else:
+            ret[current_filename].append(line)
+    return ret
+
+def _get_filenames(line, src_prefix, dst_prefix):
+    """
+    Extracts the source and destination filenames from a diff line.
+    """
+    src_index = line.find(src_prefix)
+    dst_index = line.find(dst_prefix)
+
+    if src_index == -1 or dst_index == -1:
+        return None, None
+
+    src_filename = line[src_index + len(src_prefix) - 1:dst_index - 1].strip()
+    dst_filename = line[dst_index + len(dst_prefix) - 1:].strip()
+
+    return src_filename, dst_filename
+
+def _render_diff(content, template, template_folder, **kwargs):
+    from conan import __version__
+    template = Template(template, autoescape=True)
+    def _safe_filename(filename):
```

Comment:
This function and the few below it are meant to hide the full path and implementation details of the references